app/main.py

```
@@ -1,3 +1,56 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: object) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: object) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: object) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: object) -> Distance:
+        return Distance(round(self.km / other, 2))
```

Issue: The __truediv__ method does not handle division with another Distance instance as required. According to the checklist, division should support both Distance / Distance (returning a float) and Distance / number (returning a Distance), and should raise a TypeError for unsupported types. Please update this method to check the type of 'other' and handle both cases appropriately.